File: Resource/51900286-51900712-51900718-51900699/source demo/Web Application/ComNha_API/ComNha_API/Models/Nhanvien.cs
using System;
using System.Collections.Generic;

namespace ComNha_API.Models
{
    public partial class Nhanvien
    {
        public Nhanvien()
        {
            Hoadons = new HashSet<Hoadon>();
            Lichhens = new HashSet<Lichhen>();
            Phieugoimons = new HashSet<Phieugoimon>();
            Phieunhaps = new HashSet<Phieunhap>();
            Taikhoans = new HashSet<Taikhoan>();
        }

        public string MaNhanVien { get; set; } = null!;
        public string TenNhanVien { get; set; } = null!;
        public string SoDienThoai { get; set; } = null!;
        public double Luong { get; set; }
        public string ChucVu { get; set; } = null!;
        public string DiaChi { get; set; } = null!;
        public string Email { get; set; } = null!;

        public virtual ICollection<Hoadon> Hoadons { get; set; }
        public virtual ICollection<Lichhen> Lichhens { get; set; }
        public virtual ICollection<Phieugoimon> Phieugoimons { get; set; }
        public virtual ICollection<Phieunhap> Phieunhaps { get; set; }
        public virtual ICollection<Taikhoan> Taikhoans { get; set; }
    }
}

File: Dev/51900718/src/main/java/tdtu/advanced/java/thinh68/models/NhanVien.java
package tdtu.advanced.java.thinh68.models;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Table(name = "nhan_vien")
public class NhanVien {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ma_nhan_vien")
	private Long maNhanVien;

	@Column(name = "ten_nhan_vien")
	private String tenNhanVien;

	@Column(name = "so_dien_thoai")
	private String soDienThoai;

	@Column(name = "luong")
	private double luong;

	@Column(name = "chuc_vu")
	private String chucVu;

	@Column(name = "dia_chi")
	private String diaChi;

	@Column(name = "anh_dai_dien")
	private String anhDaiDien;

	@Column(name = "email", unique = true)
	private String Email;

	@OneToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "nhan_vien_tai_khoan_forgein_key", referencedColumnName = "ma_tai_khoan")
	private TaiKhoan taiKhoan;

	@OneToMany(mappedBy = "nhanVien")
	private List<LichHen> nhanVien_LichHens;

	@OneToMany(mappedBy = "nhanVien")
	private List<PhieuGoiMon> nhanVien_PhieuGoiMons;

	@OneToMany(mappedBy = "nhanVien")
	private List<PhieuNhap> nhanVien_PhieuNhaps;

	@OneToMany(mappedBy = "nhanVien")
	private List<HoaDon> nhanVien_HoaDons;
}